exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/JoinPrel.java

```diff
@@ -186,6 +188,7 @@ public boolean isSemiJoin() {
       // fields, left fields, and right fields. Very similar to the
       // output row type, except that fields have not yet been made due
       // due to outer joins.
+      @SuppressWarnings("deprecation")
       RexChecker checker =
               new RexChecker(
                       getCluster().getTypeFactory().builder()
```

Comment:
Please use Builder class, as it is recommended in the JavaDoc:
```suggestion
      RexChecker checker =
              new RexChecker(
                      new RelDataTypeFactory.Builder(getCluster().getTypeFactory())
```